commands/roles.js

```diff
@@ -30,23 +70,25 @@ const addRole = async (interaction, guildSettings) => {
 };
 
 const removeRole = async (interaction, guildSettings) => {
-	const chosenRole = getChosenRole(interaction, guildSettings);
+	const roleProvided = interaction.options.getRole('role');
 
-	if (
-		!interaction.member.roles.cache.some(
-			(role) => role.name === chosenRole.name
-		)
-	) {
+	if (!isManagedRole(roleProvided)) {
+		return interaction.editReply({
+			embeds: [embed("💔 **You can't remove that role.**")],
+		});
+	}
+
+	if (!memberHasRole(interaction.member, roleProvided)) {
```

Comment:
This should be above L75 in my opinion; we should check if the user has the role before we check if we can remove it.